lib/beaker/test_suite.rb

```diff
@@ -1,407 +1,17 @@
-      @run = true
-      start_time = Time.now
-
-      #Create a run log for this TestSuite.
-      run_log = log_path("#{@name}-run.log", @options[:log_dated_dir])
-      @logger.add_destination(run_log)
-
-      # This is an awful hack to maintain backward compatibility until tests
-      # are ported to use logger.  Still in use in PuppetDB tests
-      Beaker.const_set(:Log, @logger) unless defined?( Log )
-
-      @test_suite_results.start_time = start_time
-      @test_suite_results.total_tests = @test_files.length
-
-      @test_files.each do |test_file|
-        @logger.info "Begin #{test_file}"
-        start = Time.now
-        test_case = TestCase.new(@hosts, @logger, options, test_file).run_test
-        duration = Time.now - start
-        @test_suite_results.add_test_case(test_case)
-        @test_cases << test_case
-
-        state = test_case.test_status == :skip ? 'skipp' : test_case.test_status
-        msg = "#{test_file} #{state}ed in %.2f seconds" % duration.to_f
-        case test_case.test_status
-        when :pass
-          @logger.success msg
-        when :skip
-          @logger.warn msg
-        when :fail
-          @logger.error msg
-          break if @fail_mode.to_s !~ /slow/ #all failure modes except slow cause us to kick out early on failure
-        when :error
-          @logger.warn msg
-          break if @fail_mode.to_s !~ /slow/ #all failure modes except slow cause us to kick out early on failure
-        end
-      end
-      @test_suite_results.stop_time = Time.now
-
-      # REVISIT: This changes global state, breaking logging in any future runs
-      # of the suite – or, at least, making them highly confusing for anyone who
-      # has not studied the implementation in detail. --daniel 2011-03-14
-      @test_suite_results.summarize( Logger.new(log_path("#{name}-summary.txt", @options[:log_dated_dir]), STDOUT) )
-
-      junit_file_log  = log_path(@options[:xml_file], @options[:xml_dated_dir])
-      if @options[:xml_time_enabled]
-        junit_file_time = log_path(@options[:xml_time], @options[:xml_dated_dir])
-        @test_suite_results.write_junit_xml( junit_file_log, @options[:xml_time] )
-        @test_suite_results.write_junit_xml( junit_file_time, @options[:xml_file], true )
-      else
-        @test_suite_results.write_junit_xml( junit_file_log )
-      end
-      #All done with this run, remove run log
-      @logger.remove_destination(run_log)
-
-      # Allow chaining operations...
-      return self
-    end
-
-    #Execute all the TestCases in this suite.
-    #This is a wrapper that catches any failures generated during TestSuite::run.
-    def run_and_raise_on_failure
-      begin
-        run
-        return self if @test_suite_results.success?
-      rescue => e
-        #failed during run
-        report_and_raise(@logger, e, "TestSuite :run_and_raise_on_failure")
+    def self.runner(runner)
+      case runner
+      when "beaker"
+        ::Beaker::Runner::Native::TestSuite
```

Comment:
Why Native for the module wrapping TestCase?  Is this because other runners will have their own notion on what a TestCase is, so we're keeping these things in the Native namespace?

If that's the case, then I think this is an opportunity to clear up potential future confusion.  Having the runner be named Beaker means it'll easily be conflated with Beaker in general.  With the Native module, there's also a mapping that'll have to be maintained in people's understanding that the "beaker runner" is equivalent to the Native module in the code.  If we have this string (& the preset) named "native", then we eliminate both the confusion possibility, & stay consistent with our module path.